sn_client/src/api.rs

```diff
@@ -340,10 +340,18 @@ impl Client {
         address: XorName,
         wallet_client: &mut WalletClient,
         verify_store: bool,
+        public: bool,
     ) -> Result<(ClientRegister, NanoTokens, NanoTokens)> {
         info!("Instantiating a new Register replica with address {address:?}");
-        let (reg, mut total_cost, mut total_royalties) =
-            ClientRegister::create_online(self.clone(), address, wallet_client, false).await?;
+        let (reg, mut total_cost, mut total_royalties) = match public {
+            false => {
+                ClientRegister::create_online(self.clone(), address, wallet_client, false).await?
```

Comment:
better rename to `create_private_online` to be more explicit ?

Reply:
agreed - it was an existing name and I was hesitant to change it, but will per your suggestion

Reply:
Just pushed a change, since I was changing the existing function anyway, I opted for adding Permissions argument and combining create_online and create_public_online into a single function (since they differed only by the permission).